### PR TITLE
build for release, including exporting artefacts and fixtures

### PR DESCRIPTION
**Pull request by chrismaree**

### CODE REVIEW

File: test/chain-specific-spokepools/Arbitrum_SpokePool.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ describe("Arbitrum Spoke Pool", function () {
     l2GatewayRouter = await createFake("L2GatewayRouter");
 
     arbitrumSpokePool = await (
-      await getContractFactory("Arbitrum_SpokePool", { signer: owner })
```
Comment (chrismaree) [inline test/chain-specific-spokepools/Arbitrum_SpokePool.ts:27]: this syntax was actually redundant all over the place and can be removed without impacting anything.

File: test/utils.ts [hunk 2]
```diff
@@ -63,6 +70,12 @@ function getArbitrumArtifact(contractName: string) {
   }
 }
 
+// Fetch the artifact from the publish package's artifacts directory.
+function getLocalArtifact(contractName: string) {
```
Comment (chrismaree) [inline test/utils.ts:74]: this method enables the util to find artefacts when run from node modules.

File: test/utils.ts [hunk 3]
```diff
@@ -27,22 +27,29 @@ export async function getContractFactory(
...
           const arbitrumArtifact = getArbitrumArtifact(name);
           return new ContractFactory(arbitrumArtifact.abi, arbitrumArtifact.bytecode, signerOrFactoryOptions as Signer);
-        } catch (error) {
-          throw new Error(`Could not find the artifact for ${name}!`);
+        } catch (_) {
+          // Finally, try importing the package from the local path. This would be the case when running these utils
+          // from node modules which breaks using the hardhat getContractFactory function.
+          try {
+            const localArtifact = getLocalArtifact(name);
+            return new ContractFactory(localArtifact.abi, localArtifact.bytecode, signerOrFactoryOptions as Signer);
```
Comment (mrice32) [inline test/utils.ts:49]: So basically this won't work with libraries?

File: test/utils.ts [hunk 4]
```diff
@@ -63,6 +70,12 @@ function getArbitrumArtifact(contractName: string) {
   }
 }
 
+// Fetch the artifact from the publish package's artifacts directory.
+function getLocalArtifact(contractName: string) {
+  const artifactsPath = `${__dirname}/../../artifacts/contracts`;
+  return findArtifactFromPath(contractName, artifactsPath);
```
Comment (mrice32) [inline test/utils.ts:76]: This might be a bit inefficient, but we could always add caching later to speed things up.

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**